vhost/src/lib.rs

```diff
@@ -51,6 +51,9 @@ pub mod vhost_user;
 #[cfg(feature = "vhost-vsock")]
 pub mod vsock;
 
+#[cfg(all(feature = "postcopy", feature = "xen"))]
+compile_error!("Both `postcopy` and `xen` features can not be enabled at the same time.");
```

Comment:
I'd document this in the README (in the `Xen support` section), explaining also why.
Maybe we can add also a little comment here with the reason behind this incompatibility.

Reply:
Added new postcopy section with explanation. Also added comments above the compile check.